Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -219,31 +219,46 @@ void PeakFitPreCheckResult::setNumberOfPeaksWithNotEnoughDataPoints(const size_t
 
 void PeakFitPreCheckResult::setNumberOfPeaksWithLowSignalToNoise(const size_t n) { m_low_snr = n; }
 
-void PeakFitPreCheckResult::report(Kernel::Logger &logger) {
+bool PeakFitPreCheckResult::isIndividualPeakRejected() const {
+  // the method should be used on an individual peak, not on a spectrum
+  assert(m_submitted_spectrum_peaks == 0);
+  assert(m_submitted_individual_peaks == 1);
+
+  // if a peak is rejected, it is rejected based on the very first check it fails
+  size_t individual_rejection_count = m_low_count_individual + m_not_enough_datapoints + m_low_snr;
+  assert(individual_rejection_count <= 1);
```

Comment:
I think these `assert` statements only apply on debug builds - is that what you intended?

Reply:
Correct, it's a self-consistency check for debug only, to make sure the class is used as designed.